src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/ElasticSearchOutput.cs

```diff
@@ -19,17 +21,43 @@
 using Validation;
 using RequestData = Microsoft.Diagnostics.EventFlow.Metadata.RequestData;
 
+[assembly: InternalsVisibleTo("Microsoft.Diagnostics.EventFlow.Outputs.Tests")]
 namespace Microsoft.Diagnostics.EventFlow.Outputs
 {
     public class ElasticSearchOutput : IOutput
     {
-        private const string Dot = ".";
+        private static readonly GregorianCalendar Gc = new GregorianCalendar();
+
         private const string Dash = "-";
 
-        private ElasticSearchConnectionData connectionData;
+        protected internal ElasticSearchConnectionData connectionData;
 
         private readonly IHealthReporter healthReporter;
 
+        private enum FormatIndexType {QuarterOfYear, WeekOfMonth, WeekOfYearDoubleDigit, WeekOfYear}
+
+        private class FormatIndex: IComparable
+        {
+            public FormatIndex(int index, FormatIndexType formatIndexType)
+            {
+                Index = index;
+                FormatIndexType = formatIndexType;
+            }
+
+            public int Index { get; set; }
+
+            public FormatIndexType FormatIndexType { get; set; }
+
+            public int CompareTo(object obj)
```

Comment:
One last minor thing: suggest use `IComparable<T>` instead of vanilla `IComparable` and return 1 if the passed object is null, see [this example](https://learn.microsoft.com/en-us/dotnet/api/system.icomparable-1?view=net-7.0#examples).

Reply:
You know, scratch that. You put enough work into this. LMK help.